DatadogInternal/Sources/Telemetry/Telemetry.swift

```diff
@@ -173,21 +173,33 @@ extension Telemetry {
     ///   - attributes: Custom attributes attached to the log (optional).
     ///   - file: The current file name.
     ///   - line: The line number in file.
-    public func debug(_ message: String, attributes: [String: Encodable]? = nil, file: String = #file, line: Int = #line) {
+    public func debug(_ message: String, attributes: [String: Encodable]? = nil, file: String = #fileID, line: Int = #line) {
         debug(id: "\(file):\(line):\(message)", message: message, attributes: attributes)
     }
 
     /// Collect execution error.
     ///
     /// - Parameters:
     ///   - message: The error message.
+    ///   - kind: The kind of error.
     ///   - stack: The stack trace.
     ///   - file: The current file name.
     ///   - line: The line number in file.
-    ///   - file: The current file name.
-    ///   - line: The line number in file.
-    public func error(_ message: String, kind: String? = nil, stack: String? = nil, file: String = #file, line: Int = #line) {
-        error(id: "\(file):\(line):\(message)", message: message, kind: kind, stack: stack)
+    public func error(_ message: String, kind: String? = nil, stack: String? = nil, file: String = #fileID, line: Int = #line) {
+        func defaultKind(fileID: String) -> String {
+            // e.g. `DatadogInternal/Telemetry.swift` -> `DatadogInternal/TelemetryError`
+            return "\(fileID.hasSuffix(".swift") ? String(fileID.dropLast(6)) : fileID)Error"
```

Comment:
/question I don't see the benefit of doing this format, having `kind: "fileID"` and `stack: "\(fileID):\(line)"` will give us the same info and cardinality. We could avoid doing string parsing/formatting, unless I'm missing something?

Reply:
Agree, it isn't necessary 👍. ✅ Changed to `kind: #fileID`.